text/3759-cargo-supported-targets.md

```diff
@@ -0,0 +1,592 @@
+
+The `per-package-target` nightly feature defines the `force-target` field, which is supposed to
+force the package to build for a specific target-tuple. This does not interact well when used in
+dependencies, as one would expect a dependency to be built for the same target as the package.
+`supported-targets` supersedes `force-target` because instead of enforcing a single target, it
+enforces a set of targets.
+
+Published crates have mainly used their documentation to specify which targets they support, or they
+would leave it up to the user to infer it. Some crates also made use of compile time errors to
+ensure that `cfg` requirements are met, for example:
+```rust
+#[cfg(not(any(…)))]
+compile_error!("unsupported target cfg");
+```
+[`getrandom`](https://github.com/rust-random/getrandom/blob/9fb4a9a2481018e4ab58d597ecd167a609033149/src/backends.rs#L156-L160)
+is an example of a crate utilizing this method.
+
+In other system level languages, vendoring dependencies is a common practice, and the user would be
+responsible for ensuring that the dependencies are compatible with the target.
+
+Some higher-level languages and build tools have the ability to specify which platforms are compatible.
+- Python package has [classifiers](https://pypi.org/classifiers/) as package metadata that includes supported platforms
+- Python wheels (pre-built packages) have [platform compatibility tags](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#platform-compatibility-tags).
+    The reference explains how these are [used](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#use)
+    by installers to determine which build of a package to install.
+- `npm` allows specifying which [`os`](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#os) and
+    [`cpu`](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#cpu) a package supports. These generate an
+    error when installing a package that does not support the platform used.
+- Swift has [`package.platforms`](https://developer.apple.com/documentation/packagedescription/package/platforms), which
+    allows specifying which platforms and versions a package support (mostly for apple products e.g., `macOS`, `iOS`, `watchOS`, `tvOS`).
+- [Buck](https://buck2.build/docs/rule_authors/configurations/#target-platform-compatibility)
+    and [Bazel](https://bazel.build/reference/be/common-definitions#common.target_compatible_with)
+    both provide `target_compatible_with`.
+Some accept a string or list of strings representing the platforms, while `Buck` & `Bazel` seem to accept a
+form comparable to `cfg` in Rust.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Should we strip the `cfg` prefix from the field e.g., `supported-targets = 'target_os = "linux"'`?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+## Ensuring proper use of dependencies
```

Comment:
Strictly speaking, this feature is not blocked on this RFC.  We could do set logic between `target.*.dependencies` and `target.*.dependencies`.  When a package itself is only supported on some platforms, they could workaround the lack of `supported-targets` by putting *all* dependencies in a `target.*.dependencies`.